src/Compilers/CSharp/Test/Semantic/Semantics/PatternSubsumptionTests.cs

```diff
@@ -152,19 +149,19 @@ public static void Main()
         {
             case bool n:
                 break;
-            case ""foo"": // error: subsumed by previous case
-                break;
+            case ""foo"": // wrong type
+                break; // unreachable
         }
     }
 }";
             var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions);
             Assert.True(compilation.GetDiagnostics().HasAnyErrors());
             compilation.VerifyDiagnostics(
                 // (10,18): error CS0029: Cannot implicitly convert type 'string' to 'bool'
-                //             case "foo": // error: subsumed by previous case
+                //             case "foo": // wrong type
                 Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""foo""").WithArguments("string", "bool").WithLocation(10, 18),
                 // (11,17): warning CS0162: Unreachable code detected
-                //                 break;
+                //                 break; // unreachable
                 Diagnostic(ErrorCode.WRN_UnreachableCode, "break").WithLocation(11, 17)
```

Comment:
I'm not sure which it is, but in either case this is a cascaded diagnostic. It would probably be worth suppressing it, but it may be a bit awkward and it isn't my highest priority. I'll create an issue for this.

---
In reply to: [69164613](https://github.com/dotnet/roslyn/pull/12250#discussion_r69164613) [](ancestors = 69164613)